### DISCUSSION TITLE
Is it possible to use an array type instead of a tuple with static-length arrays?

### DISCUSSION THREAD

**Discussion by SleepingShell** [wagmi-dev/abitype#116] [Q&A]
This question I guess also extends to Wagmi given this struct is used as a function parameter.

I have a struct defined as such in my contract, with ``MAX_TOKENS`` as a constant:

```
struct DepositArgs {
    uint256[MAX_TOKENS] depositAmount;
    uint256[2] outCommitments;
    bytes[2] encryptedOutputs;
    bytes proof;
  }

function deposit(DepositArgs calldata args) external {
    ...
}
```

I am grabbing the type of the DepositArgs as follows:
```typescript
type DepositArgsStruct = AbiParametersToPrimitiveTypes<ExtractAbiFunction<typeof zkErc20ABI, 'deposit'>['inputs'][0]['components']>;
``` 

However, the code I have written for my utility functions uses ``bigint[]``. Is there a good way to continue interfacing with Wagmi, and abi-extracted types with arrays instead of tuples? It seems like some additional boilerplate is needed if, for instance, large arrays were used in the solidity code. Or is this considering undesirable as then the length of the arrays won't be compile-time checked?

**Comment by Raiden1411** [comment]
I think that the design decision to convert to tuples for fixed size arrays makes total sense here but if you want to change the way that those types infer to fixed size tuples from fixed sized solidity arrays you will need to reconfigure the types manually. Here is a example of what I mean on this  [TSPlayground](https://www.typescriptlang.org/play?#code/PTAEAEDMEsBsFMB2BDAtvAXKZAjaA6AFwGcAoeADwAcB7AJ0NAGMbFjHdpQBeUAbVIBIAN5DBAImiIqAVxLisfALoAaMeJToFocQFkAggA0A+gBUA8gGkAogDkAyuLWCJNObPmKxI7xKmF4OhRYUwBPKnhtcRl-ACYAVgA2J18NNEiscRSXF3FCcIydGMRCBOTvAF8xVXV2ZADdOU5YaHyogDdoeAB3bIl8iKjIGUQmQmhWcSEK51FcqQ9ibQEcuZzxFlRaRCRPfl8fHPX-QOCwwczi0qS+WKU+o7StTIATeFpiVv1UNxKH9YGhWicRudymR0EMwOa0eJyCyBCBSiVzKt3uzghTyBbkIAGEaKhUK10CUlhjHoDkSDEmjwUcoRCYccSqcEecgThQgFiLTyetNECkEw6OEAi9zO45GSDv0kZlOdzaQcGUcmfMWfDERcdAr4NKIRIBVEqHQaDRIP9cpT5Vy9XSclUjjVYRqznKdOw6DIxqB8SU6MgxqY9YR8AARd40T6EfR0ADm+v56SiyHjiat7ryMioCHtjsEzsNydekejfXEOMWy0L4jqDSaeBabUyiFYVGQoVwuYxeUzw1G40m01ISmwxGYrHYpGn0C29EYgNAolAoH0eAACqn0gE6GoV2voJuA+gd5YpC896u8OzL-YaC0Xq1QvoXi86HriLf79BH-lYwHQi-B8nwAITNWAgJ-UDbU-UgVzvYD8gAMWgCh4Bef8OwAJWQRA43gSDf1CJCRjGCZEEIp8AEkSko-J7EIOgpDjOjQlMbMEEvawKEYwMYzwEiB3Iy8sL1e92nQv0YDjUgKlASBTVQUAAHJOEBZTp0oWgGFARddBA1px24ODQAAHxU5SV3MgAOFczNAABGRIrNAWIABYXIAZliFy3IABl82z7KSFzEg8+yAHZYhMmyAvs6ygvMgBOZz7IcvyPPMhyHJ8rLYgCvLbKyzznKytzMsc+JopcpyCscxIisciLSsc6yKocpLctcvy6vyxrYic+zYncobvJi1z-KGtz+qSTTqHnXSClAXRAnwgAecwcAAK3gMYHJUUBNp2sZYgAPh4Q6iUIDbtt2wh9tAABreBQnNQ7bpO86ADJ3uO0pSGAAAqQG4MB31WAkhhx2EcAWkQR6r2gdk5MIGgJzod9iG2R88NAdl7GFaAqEYE1Z1aaAJMWiJ8FB0HQDDGg9VAVtGCkJhYBkN55JkWBYGwDGO1AehdI4+AJ0QSHPlYfBQAAVWIMWYbhhGDyPbdAlMGh1yYq6KfgZHdLR4ZedCcXIY4AXQnHXCXhFnM9Rp0AQadiB22PPGD3ZUAAFol1hqQVevAoUbRlgJcCBc0fxwnidAd8TT1XZ6nIunwDdtAPY3LcT0CM9EFt33zGJ8iET95XEbVnO6Dzl4Q+Z1NTW6UBOVAdOq6p+BU-fQgZCCcdo6Y2PSd1ynAVB4ByHmnTF09gpNe1snxgk9k1pM0xZ4iUBKACfPxw3gi19V7P4FPc8t54pAXj3rPjxP3Oz94I-b9P-O1HO3gF5H-WCgAGTNR7sxrXXkHCIB1gGHmPi-F4p0+DgPZEoacIA5afFxsgUAqBkBUENqAWA-9QDZg7uOaA45IDIHYIEA6hAAAWuFmYhnQuLX8Jc+aAjIEg4gwA0ZUMIIQKgxAMAgEIN0VoO58CbGAPYeAyAsKdkQLiYAdYZAcIcvETy1kkoRWys1Ty8RNEOTUfEaypBFyf3JsvX+-9AGHxAWLbel9r5IwKJecBlc77VzPnY3eFdIH33zhdJ+6t3Gv1IO-JcJk+DGFAFIUACEoJ-lfJjYgSgsCiWIOJSSrBpJ8HEC+N8H52TiAQXJH6y5+CROibEoiYF7zJObuBKRiBZKgBKeE8piAYnfiqTBWpqT0kvCktAOM2SQIwQKUoWBASq41yKc0sJK4IlRPaZUp8gkyKsFqQAAwACTCF6bACS-TMmDL4MpXJiT2TKSUBUHZnpmIVA2U0lp8y2kdMQqEGihBalwKWp4q+oBtnCGUjISy5llLKQqP4HZUhICBDxgZEg9yTIrgAPxwsMufHefznqvUgKAeFxB2R-xoAAqgSK7KovxYSyxVBYH4oQXZOyWBlKy3hq2bo7S8AkFAO0BEMh4D4A0gyrAGyWWPTZe05Syz8gfMspAegGDQwPOKXMspizXlxNCAxJieFam3Lwo8lVCyKmdKfOxe2PTdr0BeGtPVLF8GspoOy06BrSlGqWSav8ltanvmQC8VgsBTYjDFY6xAyhZLTkXAAcR9TuUwNDEDTXxRdbi7NObwDWvpQyB1GqDXMiNcy3l7KTXMtNeyYLnWLh-vk+N5g6DWAAI4yDZDQRN6LeApo5m8DN+KDrRqkbG+NrauWguUhWpatgaBJvbRQVNXbM0kF7TGjWg7rJJvMlW4gBKa11sbc2odxAx2b0pRY4lBDeCupeQCjdW7cK1obU2kILbV2GXuRatJ+yMmICycpD5FyZlPNVdEgFfb6jLtwvu19KSxIfsOV+45ykDJxl-QUS5LrWlqonXSqD76DkDKGQhwZyGIioaqADYGdM-Tm2hv7eG3jn6BDriwDGepsbMTxgUAmg8SY6zMWLQEjtnZgzTtnTOED6N0B9mXAOdHAmMYhhHbBA8iaMHjpjJOg5Gku2E+7FxPigkF0OsXVgpclbScmW4mudcUAY0dc3U2bc3Edy7ifXubB2MRE48p1uPGl58YKOPSe2kFxLXM7GrWPm9Yr2sWJwJGL7EyfbuZYQ9d0BYFtQAbg7lgIN4rQBQhXLp8TNc4teNC7422j8b6BJrm-HgJkkEORlriKhu0Eao1AArMW0BcWLmIc3Mh0AmCgAABQsztggIWEmG4dgAJTYD8UwWhOAxa4OJQwghs50CPlAwGx2onXE7hOepUcvzxwdrTWtfeB0pVsVFvZG7mFQjOoZai-e88IvmIiEAsrdAjsodUPtvTNdnuMtAEg2ITWWtMARt1juUTxw93tvN22LBzZb0DFQicc4dglDJYV2Lp2VUMpXICLAN2zUIDJQyzY2xdj8MWTCiTph8TY7pyVv5Pq-WIADQltxYbid5bJai5nBJaeknZ+OTn-rTb8+J6i2XQrQDGAABJkJZQKDCVWq5AJZ2Lkg51CeMT5VT0AqKkGq-HCMDX42xYOZ3OONJccXNBDHPzACJuVxS+5zLj3dl8D+9KQLhlfBLBquxW9EXrPSS1J++9xekWCiryD0HyPeueSWBqMngX+Opnnl9wykHyf8tB-pcTrASD9C8xt63PT1t3wpYYY77urnXc0A+oQX3geg98F16wXYarU999JHwRAMhUDLboKOMh4Mo+MEJ135PAo0uMWYvnwXWfyUz71ycgUqGN8riwDsSGMetduLj1-FeveceEDAT94HvuqiK6QZ5SHrWok9aWn16boQ9sriBs7BlMGFCOgdgA6aFWFDZZXKRF4DZbAOMZAKQdgbAJ6XLT4AALz803mG3gFaBawkxgDQheG9gQDwmoTd0Fh5Q5jFmFg2SyA2Rm1-xXAALsjBnMhog8H33MigN9Sz1Mip1YJ9m9iEOEJEJEPslEIkN9j4OJwEI2VtWUFgOJ3MjkJXzwkULsmkMAPshUO1SGXGXGU8iUFgOUPkP0OMP4IngZRzz52UmOwl3+ShS-VhW4Nrj4B2V9xHQ8IcOEBuxQkIMexwjwngHuQqCMKFzB3IyDzBjjSQDAKcIk2iWoWoKoA+FaDFkfHfDWQonwRQTjF0haxQJDX+RcNgPa3AIkySPPj4n+XsGgAwNgMoL5UYJYI9wEPYLkGT3MlqIwLMlaPEMkLEPMkGL6OUJ2RcIqAUJcloPEHMKiO0LGOgImMMOMP+U8kUM0JkMsOJ2sMO1sP+3sIBXGLcOEHKJiTqOCLCIF1RS9x51j3C3j0+3TTXxXBWnjHTR2IoSXCy1ABcLy1OkvA3w+P0zXxgVL0VyP0CDJXLzADchliog-03j6zGxwAGyYEoVFgOmFm-wOm6HSNYFsMKKbm6BoQXAKN62IEYKQTjT60W3aRoRSKQFAGJMZLQQIPQmIKQDjDIO-3hyFg6LegDCCJGw2W6PgFgL9T1EQAJMoGIVZnaRFI9WIlQnQke1FKJRJXoIBLBzAEqJ5JoXHDeGJkxzjCXQqPjX+TwxOUe10GQAoAjCNMuQ2QOhPiYEYL2VwyOXwy1UG34kcWIxOwvi8SN07he0B3Ez+39PsNOO+U3kJ1tRN1RRFQdXZTh2Uh2RjOCOUnwAeQF0P3gEhihPtWDXZUQTAAsDDHMDJwiHWywRbmbyCDY023QmgB21NmnzSVYFd2tzt1hT6zlRGBeBpkXBVzV0QA1x+2+1P3t3sJuNNh+2UHfmiwO0CHHEJ1nP2AZVON+MJx+y1P93wGjIQL5jXOgOl150O0z1ADBNNx+OgPsOSyXw61ULyMfzl1vN9R33SEuXsLBV91RWDN9ywBHOIHV3SE1xi211MCPML1BwAsV1IVgAVkQUiLBkowjjrwAiFlxVM1ox+zk2YyxlYBxjyKUyHg+0wIdlpi0zbjDMCXHF9keywqk1wqnIY2wTDjR3a1IpU3eDUxKGTknFThoqBOK0LiM2CGYsDggos3PCswblsxbh7IqP8y03rLc0Yojw4xjm4weIorICdgni0gWhnlYpAPP143ZGICT1oqrlXMDI51PO93PMCGUGcTvw8XsocWXP038VMpqxCQulKSpPMArKwDQunjRlgFwHgAQFtkRwm0dyUobwdyoVsw23aWQDkBoBpwQACHqzAH5TjBlj4DlRoGX10IOhRLoHKuYlLyQW4V4X4RADjFwJkBwDEQJGACJGFCjHNEIGAG4uAGIWID5Q4XKiSk8iSnCVD2iXD1xSBKSSwDuNMV8yiysJ+3T0vIK3cvzhMmdVI2MSWntKjC+DTC1W9EYEq2kvt3MtWoKCsu4l4jGAPFWQ0zWkBDek4AOmUkNJOsIFHROQWClEuT4D8nGWUhpyHxIEuVOnSwBjADsgAD1kUgA). 

This however won't work if you are using wagmi's `readContract` and others alike since you can't really extend the type config like it's shown [here](https://abitype.dev/config.html). 

We could possibly create a config for this type of situation but I think that it would lead to incorrect inference if we were to support it.

**Reply by SleepingShell** [reply]
Thanks for the response.  I see the motivation for the design and it's probably the best and safest approach, I'll adopt accordingly.

Btw, could you confirm if the way I am grapping the ``depositArgsStruct`` type is fine?

**Reply by Raiden1411** [reply]
As long as it gets you the type you want its good. If you want something more generic consider taking a look at the examples/types.ts or viem's exported type helpers
